packages/astro/src/runtime/server/hydration.ts

```diff
@@ -2,12 +2,10 @@ import type { AstroComponentMetadata, SSRLoadedRenderer } from '../../@types/ast
 import type { SSRElement, SSRResult } from '../../@types/astro';
 import { hydrationSpecifier, serializeListValue } from './util.js';
 import { escapeHTML } from './escape.js';
-import serializeJavaScript from 'serialize-javascript';
 
 // Serializes props passed into a component so that they can be reused during hydration.
-// The value is any
 export function serializeProps(value: any) {
-	return serializeJavaScript(value);
+	return JSON.stringify(value);
```

Comment:
You can use [SuperJSON](https://github.com/blitz-js/superjson)

Reply:
Way too big for this use case IMO.

Reply:
I'm torn on this one since it removes functionality, but maybe it is the right move to go with `JSON.stringify()` here

> functions, regexps, dates, sets or maps

This is the list of datatypes that `serialize-javascript` helps with.  Functions are the only ones I could see being used in a component, but then maybe that really isn't an ideal use case anyway? 🤔

Reply:
Removed the dep, thanks!

Reply:
> Way too big for this use case IMO

Well point. And what about [devalue](https://github.com/Rich-Harris/devalue)? Sorry for insisting, but not having `undefined`, `Date` and may break existing code

Reply:
No worries! that's definitely a better option. I still don't feel super great about putting it into a script tag but it's better than having many script tags. Let me try that.

Reply:
Ok, I have Tried this out. It turns the inline script from 254 bytes to 2.08kB :(

Reply:
> It turns the inline script from 254 bytes to 2.08kB :(

How weird. devalue itself shouldn't be included in the output

Reply:
I just mean that devalue adds that many bytes. I don't feel comfortable adding that in a PR to fix a regression. Maybe it's a good tradeoff and maybe not, but I think we should weight the pros and cons in a separate issue. So I'm going to revert the `astro-island` change and go back to script tags until we have time to figure out what to do here.